addons/resource/models/resource.py

```diff
@@ -693,6 +686,24 @@ def create(self, values):
                 values['tz'] = tz
         return super(ResourceResource, self).create(values)
 
+    def write(self, values):
+        if 'company_id' in values and not values.get('calendar_id'):
+            values['calendar_id'] = self._get_calendar_for_company(values.get('company_id'))
+        return super().write(values)
+
+    @api.model
+    def _get_calendar_for_company(self, company_id=False):
+        if company_id:
+            return self.env['res.company'].browse(company_id).resource_calendar_id.id
+        else:
+            calendar = self.env['resource.calendar'].search([('company_id', '=', False)], limit=1)
+            if not calendar:
+                calendar = self.env['resource.calendar'].create({
+                    'name': _('Standard 40 hours/week'),
+                    'company_id': False,
+                })
```

Comment:
will globally crash because users globally do not have the rights to create `resource.calendar` records